slack_bolt/middleware/function_token/function_listener_token.py

```diff
@@ -0,0 +1,24 @@
+from typing import Callable
+
+from slack_bolt.request import BoltRequest
+from slack_bolt.response import BoltResponse
+from slack_bolt.middleware.middleware import Middleware
+
+
+class FunctionToken(Middleware):  # type: ignore
+    def process(
+        self,
+        *,
+        req: BoltRequest,
+        resp: BoltResponse,
+        # As this method is not supposed to be invoked by bolt-python users,
+        # the naming conflict with the built-in one affects
+        # only the internals of this method
+        next: Callable[[], BoltResponse],
+    ) -> BoltResponse:
+        if req.context.bot_access_token:
+            req.context.client.token = req.context.bot_access_token
+            return next()
+
+        # As the token was not found for this function call, skip running the listener
```

Comment:
Is it safe to assume that bot_access_token always exists for function events? If not, we still should call `next()` in any case here.

Reply:
I think the only situation where `bot_access_token` is not present is when this middleware is used on a _none function related request_ in this case it could just handle the listener normally, Ill add this change ☄️